### PR TITLE
[Merged by Bors] - Add Info request to post client

### PR DESCRIPTION
**Pull request by fasmat**
## Motivation
Part of #5149 

This extends the `PostClient` to be able to fetch metadata about the PoST from the service

## Changes
- Add method to fetch metadata from PoST service

## Test Plan
- Tests were added for new functionality with and without TLS

## TODO

- [x] Explain motivation or link existing issue(s)
- [x] Test changes and document test plan
- [x] Update documentation as needed
- [x] Update [changelog](../CHANGELOG.md) as needed

### CODE REVIEW

File: api/grpcserver/post_client.go [hunk 1]
```diff
@@ -29,82 +29,132 @@ func newPostClient(con chan<- postCommand) *postClient {
...
-			}
+		case <-time.After(2 * time.Second):
+			// TODO(mafa): make polling interval configurable
+			continue
 		}
 	}
+
+	proof := proofResp.GetProof()
+	metaData := proofResp.GetMetadata()
+	if metaData == nil {
```
Comment (poszu) [inline api/grpcserver/post_client.go]: This looks weird. How about:
```suggestion
	metadata := proofResp.GetMetadata()
	if metadata == nil {
```

File: api/grpcserver/post_service_test.go [hunk 2]
```diff
@@ -233,6 +233,86 @@ func Test_Cancel_GenerateProof(t *testing.T) {
...
+	require.Eventually(t, func() bool {
+		meta, err = client.Info(context.Background())
+		return err != nil
+	}, 5*time.Second, 100*time.Millisecond)
+
+	require.ErrorContains(t, err, "post client closed")
+	require.Nil(t, meta)
+}
+
+func Test_Metadata_TLS(t *testing.T) {
```
Comment (poszu) [inline api/grpcserver/post_service_test.go]: Is this test case really needed? TLS is orthogonal to the grpc method functionality, isn't? I mean that it's sufficient to test if TLS works in isolation and then test all methods just in "plain mode".

**Review by poszu — APPROVED** [review]

File: api/grpcserver/post_service_test.go (hunk 2 above)
Comment (fasmat) [reply api/grpcserver/post_service_test.go]: It doesn't hurt to have both, the test only takes ~ 3 seconds to execute. But sure I removed it.

**Comment by fasmat** [comment]
bors merge

**Comment by fasmat** [comment]
bors merge

**Pull request closed without merging**